dartagnan/src/main/java/com/dat3m/dartagnan/analysis/graphRefinement/graphs/eventGraph/utils/MaterializedGraph.java

```diff
@@ -0,0 +1,125 @@
+package com.dat3m.dartagnan.analysis.graphRefinement.graphs.eventGraph.utils;
+
+import com.dat3m.dartagnan.analysis.graphRefinement.graphs.eventGraph.AbstractEventGraph;
+import com.dat3m.dartagnan.analysis.graphRefinement.util.EdgeDirection;
+import com.dat3m.dartagnan.utils.timeable.Timestamp;
+import com.dat3m.dartagnan.verification.model.Edge;
+import com.dat3m.dartagnan.verification.model.EventData;
+import com.dat3m.dartagnan.verification.model.ExecutionModel;
+
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+/*
+    A meterialized graph simply encapsulates a SimpleGraph and delegates all its methods
+    to the underlying SimpleGraph.
+ */
```

Comment:
Why do we need this wrapper then?

Reply:
Because there are cases where you want to override some specific methods because they could be handled more efficiently. Basically, it is a "forwarding decorator".